### PR TITLE
Ad7124pmdz demo dev

### PR DESCRIPTION
**Pull request by adrimbarean**
No-OS project for ad7124-PMOD.

### CODE REVIEW

File: projects/ADuCM3029_demo_ad7124_8PMDZ/src/app.c [hunk 1]
```diff
@@ -0,0 +1,1806 @@
...
+{
+	uint8_t reg_addr, i;
+	struct ad7124_st_reg *reg_struct_ptr;
+	int32_t ret;
+	uint8_t buff[50];
+
+	reg_addr = strtol((char *)arg, NULL, 16);
+
+	for(i = 0; i < AD7124_REG_NO; i++) {
+		if(dev->ad7124_handler->regs[i].addr == reg_addr) {
```
Comment (mchindri) [inline projects/ADuCM3029_demo_ad7124_8PMDZ/src/app.c:255]: I think you shouldn't access what is inside dev.
A better approach will be to iterate throught an array like this: https://github.com/analogdevicesinc/no-OS/blob/0a574cbbbc092523741b9fd8c7f6ff88d9770597/drivers/adc/ad7124/ad7124_regs.c#L44

File: projects/ADuCM3029_demo_ad7124_8PMDZ/src/app.c [hunk 2]
```diff
@@ -0,0 +1,1806 @@
...
+	if(i == AD7124_REG_NO)
+		return cli_write_string(dev->cli_handler,
+					    (uint8_t*)"Wrong register address.\n");
+	if(reg_struct_ptr->rw == AD7124_R)
+		return cli_write_string(dev->cli_handler,
+					    (uint8_t*)"Register is read only.\n");
+
+	reg_val = strtol((char *)reg_val_ptr, NULL, 16);
+
+	reg_struct_ptr->value = reg_val;
```
Comment (mchindri) [inline projects/ADuCM3029_demo_ad7124_8PMDZ/src/app.c:309]: by modifying this you modify the internal structure of ad7124_8pmdz_dev and an app shouldn't do that.

File: projects/ADuCM3029_demo_ad7124_8PMDZ/src/app.c (hunk 1 above)
Comment (adrimbarean) [reply projects/ADuCM3029_demo_ad7124_8PMDZ/src/app.c:255]: That type of array needs to be attached to the handler in the setup routine.

File: projects/ADuCM3029_demo_ad7124_8PMDZ/src/app.c (hunk 2 above)
Comment (adrimbarean) [reply projects/ADuCM3029_demo_ad7124_8PMDZ/src/app.c:309]: I agree, but modifying the driver is out of scope for this project.

File: projects/ADuCM3029_demo_ad7124_8PMDZ/src/app.c [hunk 3]
```diff
@@ -0,0 +1,1806 @@
...
+	uint8_t reg_addr, i;
+	struct ad7124_st_reg *reg_struct_ptr;
+	int32_t ret;
+	uint8_t buff[50];
+
+	reg_addr = strtol((char *)arg, NULL, 16);
+
+	for(i = 0; i < AD7124_REG_NO; i++) {
+		if(dev->ad7124_handler->regs[i].addr == reg_addr) {
+			reg_struct_ptr = &dev->ad7124_handler->regs[i];
```
[the remark this answers is not in the document]
Comment (adrimbarean) [reply projects/ADuCM3029_demo_ad7124_8PMDZ/src/app.c:256]: Will do.

**Comment by adrimbarean** [comment]
V1 change log:
 - remove unnecessary address search;
 - fix travis complaints.

**Comment by adrimbarean** [comment]
V2 change log:
 - remove stream command clutter characters (by request);
 - fix bug where the output data rate was not set correctly for all the channels.

**Review by mchindri — APPROVED** [review]

**Pull request merged**